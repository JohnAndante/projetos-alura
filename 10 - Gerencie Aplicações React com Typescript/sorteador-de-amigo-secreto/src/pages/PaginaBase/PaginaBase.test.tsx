import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import PaginaBase from './PaginaBase';

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao,
    };
});

describe('Testes do componente PaginaBase', () => {

    test('A pÃ¡gina deve ser renderizada corretamente', () => {
        const { container } = render(
            <RecoilRoot>
                <PaginaBase />
            </RecoilRoot>
        )

        expect(container).toMatchSnapshot();
    });

});
