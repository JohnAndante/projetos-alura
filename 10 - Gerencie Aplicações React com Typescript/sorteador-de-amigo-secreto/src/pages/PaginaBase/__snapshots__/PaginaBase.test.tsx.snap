// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do componente PaginaBase A página deve ser renderizada corretamente 1`] = `
<div>
  <header
    class="cabecalho"
  >
    <div
      aria-label="Logo do Sorteador"
      class="imagem-logo"
      role="img"
    />
    <img
      alt="Participante com um presente na mão"
      class="participante"
      src="/imagens/participante.png"
    />
  </header>
  <div
    class="card"
  >
    <form>
      <div>
        <input
          placeholder="Insira os nomes dos participantes"
          type="text"
          value=""
        />
        <button
          disabled=""
        >
          Adicionar
        </button>
      </div>
      
    </form>
    <ul />
  </div>
  <footer>
    <button
      disabled=""
    >
      Iniciar brincadeira
    </button>
  </footer>
</div>
`;
